monitor/cli.py

```diff
@@ -329,5 +303,136 @@ def send_report(
         raise
 
 
+@monitor.command()
+@click.option(
+    '--days-back', '-d',
+    type=int,
+    default=7,
+    help="Set a limit to how far back Elementary should collect dbt and Elementary results while generating the report"
+)
+@click.option(
+    '--config-dir', '-c',
+    type=str,
+    default=os.path.join(expanduser('~'), '.edr'),
+    help="Global settings for edr are configured in a config.yml file in this directory "
+         "(if your config dir is HOME_DIR/.edr, no need to provide this parameter as we use it as default)."
+)
+@click.option(
+    '--profiles-dir', '-p',
+    type=str,
+    default=os.path.join(expanduser('~'), '.dbt'),
+    help="Specify your profiles dir where a profiles.yml is located, this could be a dbt profiles dir "
+         "(if your profiles dir is HOME_DIR/.dbt, no need to provide this parameter as we use it as default).",
+)
+@click.option(
+    '--update-dbt-package', '-u',
+    type=bool,
+    default=False,
+    help="Force downloading the latest version of the edr internal dbt package (usually this is not needed, "
+         "see documentation to learn more)."
+)
+@click.option(
+    '--profile-target', '-t',
+    type=str,
+    default=None,
+    help="if you have multiple targets for Elementary, optionally use this flag to choose a specific target"
+)
+@click.option(
+    '--executions-limit', '-el',
+    type=int,
+    default=30,
+    help='Set the number of invocations shown for each test in the "Test Runs" report'
+)
+@click.option(
+    '--disable-passed-test-metrics',
+    type=bool,
+    default=False,
+    help="If set to true elementary report won't show data metrics for passed tests (this can improve report creation time)."
+)
+@click.option(
+    '--update-bucket-website',
+    type=bool,
+    default=True,
+    help="Update the bucket's static website with the latest report"
+)
+@click.option(
+    '--aws-profile-name',
+    type=str,
+    default='default',
+    help="AWS profile name for AWS",
+)
+@click.option(
+    '--aws-access-key-id',
+    type=str,
+    default=None,
+    help="The access key id for AWS"
+)
+@click.option(
+    '--aws-secret-access-key',
+    type=str,
+    default=None,
+    help="The secret access key for AWS"
+)
+@click.option(
+    '--s3-bucket-name',
+    type=str,
+    default=None,
+    help="The name of the S3 bucket to upload the report to"
+)
+@click.option(
+    '--google-service-account-path',
+    type=str,
+    default=None,
+    help="The path to the Google service account json file"
+)
+@click.option(
+    '--gcs-bucket-name',
+    type=str,
+    default=None,
+    help="The name of the GCS bucket to upload the report to"
+)
+@click.pass_context
+def upload_report(
```

Comment:
maybe it will be a good thing to merge between send report and upload report?
uploading report to slack is much like uploading it to s3/gcs?

Reply:
This was my intuition as well.
Thinking it over, we got into a relatively complex option / decision matrix regarding what to do relative to the arguments that the user supplied. For instance, if he provided both S3 and Slack, then send the report to S3 and a link to the Slack, etc. But that creates a lot of complication, both in terms of development and for the user.

However, if we dumb it down so that you basically just send off the report to whichever method specified (Slack, S3, GCS) - I don't see why not and I think it'd be for the better. I also think the difference is too small to justify a separation.

The only part is the `--update-bucket-website` flag which might make _a bit_ less sense.

What do you think? @oravi @Maayan-s

Reply:
I don't think that it makes sense to implement both in send-report. My argument is that send-report is not just uploading a report to slack, it will contain more information that will be added down the road like the summary report we have talked about. Also, I think that commands with too many arguments are confusing

Reply:
I don't see how adding a summary appended to the report changes it.
It's just as if I would've uploaded the object to S3 with [metadata](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html).
Those are just different mediums of sending the report to.

Regarding the too many arguments:
1. You could also claim the same for `upload-report` if we'd add more platforms, it's inevitable.
2. Realistically, every organization will have their favored method and they'll use only that.
3. We provide the config file for that.

Reply:
I agree with @elongl.

In addition, although summary will be sent through slack (for now), it doesn't mean it will always be like this, and that it is only relevant for sending the report to slack.
I can see the following flow:
run report -> upload to s3 -> notify on slack with the summary that the report is updated (and tag the relevant users).

I think that bundling between the report on slack to the summary is bad, and summary should be a feature by its own that we can combine with anything we would like (alerts + summary, slack report + summary, s3 + summary, only summary, etc).